src/main/java/org/apache/maven/plugins/dependency/tree/VerboseGraphSerializer.java

```diff
@@ -42,30 +42,370 @@
             MANAGED_SCOPE = "managedScope";
 
     public String serialize( DependencyNode root )
+    {
+        return serialize( root, "" );
+    }
+
+    public String serialize( DependencyNode root, String outputType )
     {
         Set<String> coordinateStrings = new HashSet<>();
         Map<String, String> coordinateVersionMap = new HashMap<>();
-        StringBuilder builder = new StringBuilder();
 
         // Use BFS to mirror how Maven resolves dependencies and use DFS to print the tree easily
         Map<DependencyNode, String> nodeErrors = getNodeConflictMessagesBfs( root, coordinateStrings,
                 coordinateVersionMap );
 
+        if ( "graphml".equals( outputType ) )
```

Comment:
This long chain of if's makes me wonder if what we really need is a base abstract Serializer type and subclasses for each type.

Reply:
Just committed that